cmd/embedded-cluster/reset.go

```diff
@@ -169,66 +201,85 @@ func newHostInfo(ctx context.Context) (hostInfo, error) {
 	if err != nil {
 		return currentHost, err
 	}
+	currentHost.RoleName = "worker"
 	// control plane only stff
-	if currentHost.isControlPlane() {
+	if currentHost.Status.Role == "controller" {
+		currentHost.RoleName = "controler"
 		// fetch controlNode
 		err := currentHost.getControlNodeObject(ctx)
 		if err != nil {
 			return currentHost, err
 		}
 	}
+	// try and get custom role name from the node labels
+	labels := currentHost.Node.GetLabels()
+	if value, ok := labels["kots.io/embedded-cluster-role-0"]; ok {
+		currentHost.RoleName = value
+	}
 	return currentHost, nil
 }
 
-func checkErrPrompt(err error) bool {
+func checkErrPrompt(c *cli.Context, err error) bool {
 	if err == nil {
 		return true
 	}
 	fmt.Println("-----")
 	fmt.Println(err)
 	fmt.Println("-----")
-	fmt.Println("An error has occured while trying to reset this node.")
-	fmt.Println("Continuing may leave the cluster in an unexpected state")
+	fmt.Println("An error occurred while trying to reset this node.")
+	fmt.Println("Continuing may leave the cluster in an unexpected state.")
+	if c.Bool("no-prompt") {
+		return true
+	}
 	return prompts.New().Confirm("Do you want to continue anyway?", false)
 }
 
 var resetCommand = &cli.Command{
 	Name: "reset",
 	Flags: []cli.Flag{
 		&cli.BoolFlag{
-			Name:   "yes-really-reset",
+			Name:   "confirm",
+			Hidden: true,
+			Value:  false,
+		},
+		&cli.BoolFlag{
+			Name:  "no-prompt",
+			Usage: "Do not prompt user when it is not necessary",
+			Value: false,
+		},
+		&cli.BoolFlag{
+			Name:   "force",
```

Comment:
Remind me what exactly force does. Looks like it skips all the checks that it's safe to remove the node. Is that right? And in what situation might someone use that as opposed to `--confirm`?

Reply:
correct, `--force` skips all checks and does not attempt to contact the cluster for clean removal, this flag provides the ability to reset nodes in the following scenarios:

- the control-plane no longer exists
- the node has no internet connectivity
- the cluster is in such a state where a clean exit is not possible